sympy/polys/tests/test_rootoftools.py

```diff
@@ -138,12 +138,11 @@ def test_CRootOf___eval_Eq__():
     r1 = rootof(eq, 1)
     assert Eq(r, r1) is S.false
     assert Eq(r, r) is S.true
-    assert Eq(r, x) is S.false
+    assert Eq(r, x).lhs is r and Eq(r, x).rhs is x
     assert Eq(r, 0) is S.false
     assert Eq(r, S.Infinity) is S.false
     assert Eq(r, I) is S.false
-    assert Eq(r, f(0)) is S.false
-    assert Eq(r, f(0)) is S.false
+    assert Eq(r, f(0)).lhs is r and Eq(r, f(0)).rhs is f(0)
```

Comment:
Is it guaranteed that `is` will do the right thing here? Is `f(0)` guaranteed to be a singleton?

Reply:
This is the type of test that I wrote [this function](https://github.com/sympy/sympy/pull/7729/files#diff-15db628bd3e2f4c63dddf4d41083c2f6R165) for; it would be used like:
```python
unchanged(Eq, r, f(0))
```

Reply:
That would be useful but wasn't include...

Maybe this test for now should just be:
```python
eq = Eq(r, f(0))
assert isinstance(eq, Eq) and eq.args == (r, f(0))
```

Reply:
I think using `is` is ok here since `is` tells if the two operands are the same object...which they are. In this case they are not something like S.One and Python 1. If this is an issue it can be fixed (and should maybe be part of code quality testing).

Reply:
@smichr this only works because of caching otherwise `f(0) is not f(0)`:
```console
$ cat t.py 
from sympy import *
f = Function('f')
print(f(0) is f(0))
$ python t.py 
True
$ SYMPY_USE_CACHE=no python t.py 
False
```